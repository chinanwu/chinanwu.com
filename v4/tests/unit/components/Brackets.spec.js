import { shallowMount, mount } from "@vue/test-utils";
import Brackets from "@/components/Brackets.vue";

describe("Brackets", () => {
  it("renders", () => {
    const wrapper = shallowMount(Brackets);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders with content", () => {
    const wrapper = mount(Brackets, {
      slots: {
        content: "<div id='test'>Here</div>"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.find("#test").text()).toEqual("Here");
  });
});
