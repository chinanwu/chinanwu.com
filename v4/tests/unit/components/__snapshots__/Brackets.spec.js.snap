// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brackets renders 1`] = `
<div
  class="Brackets"
>
  <div
    class="Bracket Bracket__upper"
    role="presentation"
  />
  
  
  <div
    class="Bracket Bracket__lower"
    role="presentation"
  />
</div>
`;

exports[`Brackets renders with content 1`] = `
<div
  class="Brackets"
>
  <div
    class="Bracket Bracket__upper"
    role="presentation"
  />
  
  
  <div
    id="test"
  >
    Here
  </div>
  
  
  <div
    class="Bracket Bracket__lower"
    role="presentation"
  />
</div>
`;
